test/test-kitchen.ts

```diff
@@ -22,18 +22,33 @@ const simpleExecp = pify(cp.exec);
 const renamep = pify(fs.rename);
 const ncpp = pify(ncp.ncp);
 
+// TODO: improve the typedefinitions in @types/node. Right now they specify
+// the return type to be Error.
+interface ExecError extends Error {
```

Comment:
This is non-blocking PR commentary :)  I see the pattern of attaching a `code` property to Error objects all over our repositories.  It makes working with TypeScript and errors kind of ugly.  Two thoughts on how to address:

1. Should we maybe consider using [Error.prototype.name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name) instead?

2. Should we consider chatting with tc39 folks about adding a general purpose `code` field to the Error object?

Reply:
@jinwoo No. `ErrnoException` has a string `code`. `child_process.exec` API returns an `Error` with a number `code` property. `@types/node` doesn't have an error type for the `exec` at the moment.

Reply:
@JustinBeckwith 

> Should we maybe consider using Error.prototype.name instead?

This is not up to us. The `exec` API in Node.js core returns an `Error` with a particular structure.

> Should we consider chatting with tc39 folks about adding a general purpose code field to the Error object?

In Node.js core, James Snell is championing an effort to have a `code` property added to all errors returned by core. I think it would be worth chatting with him to the next steps beyond that.